compiler/src/main/java/org/qbicc/graph/MemberOf.java

```diff
@@ -20,7 +20,7 @@
 
         if (structureHandle instanceof PointerHandle ph && ph.getPointerValue() instanceof Deref deref) {
             /* unwrap value since this dereference should not cause an error */
-            this.structureHandle = new PointerHandle(callSite, element, line, bci, deref.getInput());
+            this.structureHandle = deref.getValueHandle();
         } else {
             this.structureHandle = structureHandle;
         }
```

Comment:
This section should be eliminated at this point and reverted back to the original simple assignment to `structureHandle`.